pkg/kubectl/cmd/top.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/renstrom/dedent"
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/kubectl"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/kubectl/metricsutil"
+)
+
+// TopOptions is the start of the data required to perform the operation. As new fields are added, add them here instead of
+// referencing the cmd.Flags()
+type TopOptions struct {
+}
+
+var (
+	topLong = dedent.Dedent(`
+		Display Resource (CPU/Memory/Storage) usage of nodes or pods.
+
+		The top command allows you to see the resource consumption of the nodes or pods.
+		It downloads the usage metrics of a given resource (node/pod) via the Resource Metrics API.
+		`)
+
+	topExample = dedent.Dedent(`
+		  # Show metrics for all nodes in the default namespace
+		  kubectl top node
+
+		  # Show metrics for all nodes in the given namespace
+		  kubectl top node --namespace=NAMESPACE
+
+		  # Show metrics for a given pod in the default namespace
+		  kubectl top pod POD_NAME
+
+		  # Show metrics for a given pod and its containers
+		  kubectl top pod POD_NAME --containers
+
+		  # Show metrics for the pods defined by the selector query
+		  kubectl top pod --selector="KEY: VALUE"`)
```

Comment:
What you have here is correct.
This comment is a high level thing to keep in mind.

Basically, we want to have kubectl to be consistent, for example all commands using the same flags (where it makes sense). --selector is on of those.
Currently you can pass --selector= (or alternatively -l). We should have exactly the same here and support exactly the same format for this flag.